### DISCUSSION TITLE
How to update the anchor in Markdown after auto translation?

### DISCUSSION THREAD

**Discussion by yanedie** [mkdocs/mkdocs#3879] [Q&A]
I wrote a Python script to translate all the markdowns under the folder docs/en-us/*  into docs/de-de/ in batch. Then I found that the markdown links weren't translated. So, the anchor (#) generated from the title is broken because the target is translated into German. Is there a good way to make it work?

**Comment by Andre601** [comment]
I feel like it would be better for you to use a plugin like [mkdocs-static-i18n](https://github.com/ultrabug/mkdocs-static-i18n) as that could ease translating to other languages for you.

I myself use it and the header links are fine there, given the translations exist as actual markdown file and are processed.
Tho, I also use the same header link for consistent linking, which I do by using the attribute list extension (`attr_list`) and then append `{ #header-link }` to the header text.

**Reply by yanedie** [reply]
Thanks. Manually assigning an anchor seems a good idea and I will consider using it.

My translation exists as an actual markdown file, too. 

I have a question. Will the plugin automatically update the header link even without appending the header link? 

Here are some background info:
``` markdown
file: home.md
---
# Hello
Hello World!
---
file: random.md
---
# How-to
Please refer to [Hello](hello#hello)
```

If I translate it into German, the paragraph will be `Please refer to [Hallo](hello#hello)`
Here the `hello` is the filename and it wouldn't change, so it is correct. But the anchor, which refers to the title, is wrong. The correct anchor should be #hallo.

**Reply by Andre601** [reply]
From what I know are header links in masked links not translated. But I can be wrong.

You can always just test this yourself and see if things work.
